pipelinewise/fastsync/commons/target_snowflake.py

```diff
@@ -154,21 +159,35 @@ def copy_to_table(self, s3_key, target_schema, table_name, is_temporary, skip_cs
                 int(skip_csv_header),
             )
         else:
-            sql = """COPY INTO {}."{}" FROM 's3://{}/{}'
-                CREDENTIALS = (aws_key_id='{}' aws_secret_key='{}')
-                FILE_FORMAT = (type='CSV' escape='\\x1e' escape_unenclosed_field='\\x1e' 
-                field_optionally_enclosed_by='\"' skip_header={} COMPRESSION='GZIP' BINARY_FORMAT='HEX')
-            """.format(
-                target_schema,
-                target_table.upper(),
-                bucket,
-                s3_key,
-                aws_access_key_id,
-                aws_secret_access_key,
-                int(skip_csv_header),
-            )
-
-        self.query(sql)
+            if self.use_iam_accesskeys:
+                aws_access_key_id = self.connection_config['aws_access_key_id']
+                aws_secret_access_key = self.connection_config['aws_secret_access_key']
+                sql = """COPY INTO {}."{}" FROM 's3://{}/{}'
+                    CREDENTIALS = (aws_key_id='{}' aws_secret_key='{}')
+                    FILE_FORMAT = (type='CSV' escape='\\x1e' escape_unenclosed_field='\\x1e' 
+                    field_optionally_enclosed_by='\"' skip_header={} COMPRESSION='GZIP' BINARY_FORMAT='HEX')
+                """.format(
+                    target_schema,
+                    target_table.upper(),
+                    bucket,
+                    s3_key,
+                    aws_access_key_id,
+                    aws_secret_access_key,
+                    int(skip_csv_header),
+                )
+                self.query(sql)
+            else:
+                sql = """COPY INTO {}."{}" FROM 's3://{}/{}'
+                    FILE_FORMAT = (type='CSV' escape='\\x1e' escape_unenclosed_field='\\x1e' 
+                    field_optionally_enclosed_by='\"' skip_header={} COMPRESSION='GZIP' BINARY_FORMAT='HEX')
+                    """.format(
+                    target_schema,
+                    target_table.upper(),
+                    bucket,
+                    s3_key,
+                    int(skip_csv_header),
+                )
+                self.query(sql)
```

Comment:
hey @mashanm , can you please resolve the conflicts? I think it's here and caused by some small changes that has been merged recently. The`self.query(sql)` calls not required here. thanks

Reply:
Addressed the conflicts and moved the query call .